# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1506017062 14400
#      Thu Sep 21 14:04:22 2017 -0400
# Node ID dbbc7e5e229474bf01e4db73c7bd7d62c08705d5
# Parent  3f036b46a1307d2bf23ee25966fcd720140b266e
avoid direct leak from fltk plot_window class

* __init_fltk__.cc (plot_window::~plot_window): Delete m_uimenu object.
(fltk_uimenu::~fltk_uimenu): Don't delete m_menubar object.

diff --git a/libinterp/dldfcn/__init_fltk__.cc b/libinterp/dldfcn/__init_fltk__.cc
--- a/libinterp/dldfcn/__init_fltk__.cc
+++ b/libinterp/dldfcn/__init_fltk__.cc
@@ -757,17 +757,17 @@ public:
   // No copying!
 
   fltk_uimenu (const fltk_uimenu&) = delete;
 
   fltk_uimenu operator = (const fltk_uimenu&) = delete;
 
   ~fltk_uimenu (void)
   {
-    delete m_menubar;
+    // FLTK is supposed to manage memory for widgets.
   }
 
 private:
 
   Fl_Menu_Bar *m_menubar;
 };
 
 #if defined (HAVE_X_WINDOWS)
@@ -890,16 +890,20 @@ public:
   plot_window (const plot_window&) = delete;
 
   plot_window& operator = (const plot_window&) = delete;
 
   ~plot_window (void)
   {
     this->hide ();
     Fl::check ();
+
+    delete m_uimenu;
+
+    // FLTK is supposed to manage memory for widgets.
   }
 
   double number (void) { return m_fp.get___myhandle__ ().value (); }
 
   void renumber (double new_number)
   {
     if (! m_canvas)
       error ("unable to renumber figure");
